src/java.desktop/share/classes/sun/awt/shell/ShellFolder.java

```diff
@@ -295,7 +296,7 @@ public static boolean isFileSystemRoot(File dir) {
      */
     public static File getNormalizedFile(File f) throws IOException {
         File canonical = f.getCanonicalFile();
-        if (f.equals(canonical)) {
+        if (f.equals(canonical) || OSInfo.getOSType() == OSInfo.OSType.WINDOWS) {
             // path of f doesn't contain symbolic links
             return canonical;
```

Comment:
Is it possible that the code below was used when the "Libraries"  or some "virtual folders" were navigated?

Reply:
Yes. I traced with debugger. 
Even after this fix, JFileChooser displays and can navigate "Libraries" and "Recent". The paths of these folders are expressed as "::{CLSID}", that can be handled correctly only with Win32ShellFolder class. And, if "Videos" is selected, the path is canonicalized. (e.g. C:\Users\User\Videos)